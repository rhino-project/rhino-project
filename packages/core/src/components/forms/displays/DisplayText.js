import { useCallback } from 'react';
import { FieldInputControlledBase } from '../fields/FieldInputControlled';
import { useGlobalComponent } from '../../../hooks/overrides';

export const DisplayTextBase = ({ empty = '-', ...props }) => {
  const accessor = useCallback(
    (value) => {
      if (!value) return empty;

      return value;
    },
    [empty]
  );

  return (
    <FieldInputControlledBase
      type="textarea"
      accessor={accessor}
      readOnly
      {...props}
    />
  );
};

export const DisplayText = (props) =>
  useGlobalComponent('DisplayText', DisplayTextBase, props);
