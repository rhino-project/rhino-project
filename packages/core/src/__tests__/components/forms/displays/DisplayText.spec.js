import { sharedDisplayTests } from './sharedDisplayTests';
import DisplayText from 'rhino/components/forms/displays/DisplayText';

describe('DisplayText', () => {
  sharedDisplayTests(DisplayText);
});
