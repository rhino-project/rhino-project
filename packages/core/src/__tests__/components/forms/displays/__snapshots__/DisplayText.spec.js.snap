// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayText > renders inside of model context 1`] = `
<DocumentFragment>
  <textarea
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
  >
    -
  </textarea>
</DocumentFragment>
`;

exports[`DisplayText > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
